src/mem/localalloc.h

```diff
@@ -90,40 +100,53 @@ namespace snmalloc
     /**
      * This initialises the fast allocator by acquiring a core allocator, and
      * setting up its local copy of data structures.
+     *
+     * If the allocator does not support lazy initialisation then this assumes
+     * that initialisation has already taken place and invokes the action
+     * immediately.
      */
     template<typename Action, typename... Args>
     SNMALLOC_SLOW_PATH decltype(auto) lazy_init(Action action, Args... args)
     {
       SNMALLOC_ASSERT(core_alloc == nullptr);
-
-      // Initialise the thread local allocator
-      init();
-
-      // register_clean_up must be called after init.  register clean up may be
-      // implemented with allocation, so need to ensure we have a valid
-      // allocator at this point.
-      if (!post_teardown)
-        // Must be called at least once per thread.
-        // A pthread implementation only calls the thread destruction handle
-        // if the key has been set.
-        handle.register_clean_up();
-
-      // Perform underlying operation
-      auto r = action(core_alloc, args...);
-
-      // After performing underlying operation, in the case of teardown already
-      // having begun, we must flush any state we just acquired.
-      if (post_teardown)
+      if constexpr (!SharedStateHandle::Options.LocalAllocSupportsLazyInit)
+      {
+        return action(core_alloc, args...);
+      }
+      else
       {
+        // Initialise the thread local allocator
```

Comment:
Is my understanding correct that this code-path gets executed on every allocation and deallocation when the CoreAllocOwnsLocalState is disabled? Since core_alloc will stay nullptr and cannot be set externally as far as I can tell? Am I missing something here?